alacritty/src/config.rs

```diff
@@ -176,8 +176,17 @@ pub fn reload_from(path: &PathBuf) -> Result<Config> {
 }
 
 fn read_config(path: &PathBuf) -> Result<Config> {
-    let mut contents = String::new();
-    File::open(path)?.read_to_string(&mut contents)?;
+    let mut bytes_vec = Vec::new();
+    let mut f = File::open(path)?;
+    f.read_to_end(&mut bytes_vec)?;
+    let mut bytes: &[u8] = &bytes_vec;
+
+    // Skip UTF-8 BOM
+    if bytes.starts_with(&[0xef, 0xbb, 0xbf]) {
+        bytes = &bytes[3..];
+    }
+
+    let contents = String::from_utf8_lossy(&bytes);
```

Comment:
We should be able to keep the current code, then add this to truncate the start of the string:
```rust
// Remove UTF-8 BOM
if buf.chars().nth(0) == Some('\u{FEFF}') {
    buf = buf.split_off(3);
}
```

Reply:
Ok 52b8d065051e0ba1bc4ff383135be6c32c2a9821